src/protocol/message.rs

```diff
@@ -208,13 +208,35 @@ named_args!(pub parse_message_set(api_version: ApiVersion)<MessageSet>,
         do_parse!(
             messages: many0!(apply!(parse_message, api_version))
          >> (MessageSet {
-                messages,
+                messages: messages.into_iter().flat_map(|i| i).collect(),
             })
         )
     )
 );
 
-named_args!(parse_message(_api_version: ApiVersion)<Message>,
+fn decompress_message(message: Message) -> Result<Vec<Message>> {
+    if message.compression == Compression::None || message.value == None {
+        return Ok(vec![message]);
+    }
+    let value = message.value.unwrap();
+    let decompressed = message.compression.decompress(&value)?;
+    let decompressed = decompressed.unwrap();
+    let version = if message.timestamp.is_some() {
```

Comment:
in fact, it should be the record format, its my misunderstand to named it to api_version.